### PR TITLE
Refactor DatabaseTablesPane and TablePane

### PR DESCRIPTION
**Pull request by calherries**
This is a brief tidying of the data reference pane in the native query editor.

image

Changes:

- TablePane: convert from .jsx to .tsx, and separated loading state (now in TableInfoLoader) from the rest of the rendering. I modelled (i.e. copy+pasted) most of `TableInfoLoader` from the [TableInfo](https://github.com/metabase/metabase/blob/eb6887733f236be0aca1f5f9ce2febf8b38a7de8/frontend/src/metabase/components/MetadataInfo/TableInfo/TableInfo.tsx#L85) component, removing everything related to rendering.
- DatabaseTablesPane: Replace loading tables with `Tables.loadList` with loading table search results from `Search.loadList`, along with models. This is a step towards add a search bar in the future.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/dataref/TablePane.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+interface TablePaneProps {
+  show: (type: string, item: unknown) => void;
+  table: Table;
+}
+
+const TablePane = ({ table, show }: TablePaneProps) => {
+  return table ? (
+    <TableInfoLoader table={table}>
+      <>
```
Comment (gusaiani) [inline frontend/src/metabase/query_builder/components/dataref/TablePane.tsx:31]: We can probably drop this Fragment, no ?
The `<>` and `<\>` on line 53?

File: frontend/src/metabase/query_builder/components/dataref/DatabaseTablesPane/DatabaseTablesPane.tsx [hunk 2]
```diff
@@ -19,40 +17,39 @@ import {
...
-    () => database.tables.sort((a, b) => a.name.localeCompare(b.name)),
-    [database.tables],
+    () => searchResults?.filter(x => x.model === "table"),
+    [searchResults],
   );
-  const sortedModels = useMemo(
-    () => models?.sort((a, b) => a.name.localeCompare(b.name)),
-    [models],
+  const models = useMemo(
+    () => searchResults?.filter(x => x.model === "dataset"),
```
Comment (gusaiani) [inline frontend/src/metabase/query_builder/components/dataref/DatabaseTablesPane/DatabaseTablesPane.tsx:34]: Search results in this iteration are no longer presented in alphabetical order.

### Now

image

### Before

image

**Review by gusaiani — CHANGES_REQUESTED** [review]
Looking good, only one thing. 

Inlined you see the spot where it looks like we are dropping the alphabetical ordering.

File: frontend/src/metabase/query_builder/components/dataref/TablePane.tsx (hunk 1 above)
Comment (calherries) [reply frontend/src/metabase/query_builder/components/dataref/TablePane.tsx]: Yes, fixed here https://github.com/metabase/metabase/pull/25751/commits/c1a2c6a5cb2e3945c37c39a00d8b9813d99c8d97

File: frontend/src/metabase/query_builder/components/dataref/DatabaseTablesPane/DatabaseTablesPane.tsx (hunk 2 above)
Comment (calherries) [reply frontend/src/metabase/query_builder/components/dataref/DatabaseTablesPane/DatabaseTablesPane.tsx]: Fixed here https://github.com/metabase/metabase/pull/25751/commits/273278be71754b4bf651d9961b406289d683b10d

Thanks for catching this!

**Review by gusaiani — COMMENTED** [review]
Thanks for the updates, Cal.

I'm seeing the failing e2e tests locally as well.

**Review by gusaiani — APPROVED** [review]
Looks good, thank you Cal.

**Pull request merged**